### DISCUSSION TITLE
Leader key not working... I think?

### DISCUSSION THREAD

**Discussion by fng97** [nix-community/nixvim#2735] [Q&A] [answered]
Hello!

Context:

Just switched from LazyVim to nixvim (yay) and have [a very minimal standalone flake](https://github.com/fng97/nixvim/blob/28c3ef9dcc18a82a7f950c86e0f073124db85cb1/flake.nix) working that also integrates with my home-manager config. I have basically no experience with manual vim configuration so please excuse me if this is a noob question 👶

System: x86-64
OS: Ubuntu (WSL2) on Windows
Shell: Fish (managed by home manager)
Nix: 2.23.0

(I am running `nix run . -- flake.nix` to test my changes) 

Problem:

I have set my leader key to space but it doesn't seem to work how I would expect it to - it just moves the cursor to the right in normal mode. I've tried installing `which-key` to get a pop-up when I press space and that doesn't work.

Tried:

- setting `mapleader` to "," in my config and running `:echo mapleader` does print a "," so the configuration is working ✅
- checking the issues and discussions in this repo for similar issues, couldn't find this raised
- checking other example configurations: everyone else just sets mapleader... am I missing some special keybind?

Edit: Added system details. Also just found `nixvim-print-init` 😍 

<details>
<summary>Click to expand nixvim-print-init output</summary>

```plaintext
nixvim on  main
❯ nixvim-print-init --version
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /nix/store/0xp0l0aj8zx6m4anxm4118nqxgxkpn4k-init.lua
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ -- Ignore the user lua configuration
   2   │ vim.opt.runtimepath:remove(vim.fn.stdpath("config")) -- ~/.config/nvim
   3   │ vim.opt.runtimepath:remove(vim.fn.stdpath("config") .. "/after") -- ~/.config/nvim/after
   4   │
   5   │ -- Nixvim's internal module table
   6   │ -- Can be used to share code throughout init.lua
   7   │ local _M = {}
   8   │
   9   │ -- Set up globals {{{
  10   │ do
  11   │     local nixvim_globals = { mapleader = " " }
  12   │
  13   │     for k, v in pairs(nixvim_globals) do
  14   │         vim.g[k] = v
  15   │     end
  16   │ end
  17   │ -- }}}
  18   │
  19   │ -- Set up options {{{
  20   │ do
  21   │     local nixvim_options = { number = true, relativenumber = true, termguicolors = true }
  22   │
  23   │     for k, v in pairs(nixvim_options) do
  24   │         vim.opt[k] = v
  25   │     end
  26   │ end
  27   │ -- }}}
  28   │
  29   │ require("gruvbox").setup({})
  30   │
  31   │ vim.cmd([[let $BAT_THEME = 'gruvbox'
  32   │
  33   │ colorscheme gruvbox
  34   │ ]])
  35   │ require("nvim-web-devicons").setup({})
  36   │
  37   │ require("which-key").setup({})
  38   │
  39   │ vim.opt.runtimepath:prepend(vim.fs.joinpath(vim.fn.stdpath("data"), "site"))
  40   │ require("nvim-treesitter.configs").setup({ parser_install_dir = vim.fs.joinpath(vim.fn.stdpath("data"), "site") })
  41   │
  42   │ require("telescope").setup({})
  43   │
  44   │ local __telescopeExtensions = {}
  45   │ for i, extension in ipairs(__telescopeExtensions) do
  46   │     require("telescope").load_extension(extension)
  47   │ end
  48   │
  49   │ require("lualine").setup({})
  50   │
  51   │ require("bufferline").setup({})
───────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by MattSturgeon** [comment, marked as answer]
I've ran into this in the past when setting `mapleader` before actually configuring any keybinds to _use_ `<leader>`.

If there's no normal-mode binding that starts with leader, then SPC will still just do the default action; which is move one column right.

**Reply by fng97** [reply]
Ah! Thanks Matt. It's working now that I've added a keybind starting with `<leader>` 🎉
